assets/js/components/HostDetails/HostDetails.jsx

```diff
@@ -79,38 +84,81 @@ function HostDetails() {
-                heartbeat={exporterStatus}
+        <div className="flex flex-wrap">
+          <div className="flex w-1/2 h-auto overflow-hidden overflow-ellipsis break-words">
+            <PageHeader>
+              Host Details: <span className="font-bold">{host.hostname}</span>
+            </PageHeader>
+          </div>
+          <div className="flex w-1/2 justify-end">
+            <div className="flex w-fit whitespace-nowrap">
+              {host.deregisterable && (
+                <CleanUpButton
+                  cleaning={host.deregistering}
+                  onClick={() => {
+                    setCleanUpModalOpen(true);
+                  }}
+                />
+              )}
+              <Button
+                type="primary-white"
+                className="inline-block mx-0.5 border-green-500 border"
+                size="small"
+                onClick={() => navigate(`/hosts/${hostID}/settings`)}
+              >
+                <EOS_SETTINGS className="inline-block fill-jungle-green-500" />{' '}
+                Check Selection
+              </Button>
+
+              <Button
+                type="primary-white"
+                className="mx-0.5 border-green-500 border disabled:bg-slate-50 disabled:text-slate-500 disabled:border-gray-400"
+                size="small"
+                onClick={() => navigate(`/hosts/${hostID}/executions/last`)}
+                disabled
+              >
+                <EOS_CLEAR_ALL
+                  className={classNames('inline-block ', {
+                    'fill-jungle-green-500': false,
+                    'fill-slate-500': true,
+                  })}
+                />{' '}
+                <span>Show Results</span>
+              </Button>
+
+              <TriggerChecksExecutionRequest
+                cssClasses="flex rounded relative ml-0.5 disabled:bg-gray-400 disabled:text-gray-200 disabled:border-gray-400"
+                disabled
               >
-                {exporterName}
-              </StatusPill>
-            )
-          )}
-          {host.deregisterable && (
-            <div className="ml-auto my-auto">
-              <CleanUpButton
-                cleaning={host.deregistering}
-                onClick={() => {
-                  setCleanUpModalOpen(true);
-                }}
-              />
+                <EOS_PLAY_CIRCLE
+                  className={classNames('inline-block mr-1', {
+                    'fill-jungle-green-500': false,
+                    'fill-gray-200': true,
+                  })}
+                />{' '}
+                <span>Start Execution</span>
+              </TriggerChecksExecutionRequest>
             </div>
-          )}
+          </div>
+          <div className="pb-3">
+            <StatusPill
+              className="self-center shadow"
+              heartbeat={host.heartbeat}
+            >
+              Agent
+            </StatusPill>
+            {Object.entries(exportersStatus).map(
```

Comment:
Just an idea, could we move this function outside ? 
Could improve readability  :)